growth/too/models.py

```diff
@@ -685,6 +689,18 @@ def flat(self):
         else:
             return self.flat_2d,
 
+    def ipix_in_greedy_confidence_interval(self, ranking):
+        """Return an array of ipix values within the greedy confidence interval
+        defined by `ranking`."""
+
+        arr = np.flatnonzero(find_greedy_credible_levels(self.flat_2d) <= ranking)
+        return psql.array(arr.tolist())
+
+    def observations_contained_within_contour(self, ranking):
+        ipix_arr = self.ipix_in_greedy_confidence_interval(ranking)
+        query = db.session.query(Observation).filter(Observation.field.ipix.overlap(ipix_arr))
+        return query.all()
```

Comment:
Let's return the query object so that the result set can be iterated over. We don't want to return the entire result set here.